.github/workflows/main.yml

```diff
@@ -10,73 +7,101 @@ on:
+      - uses: actions/cache@v3
         id: node-modules-cache
         with:
           path: |
             **/node_modules
+            /home/runner/.cache
           key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
 
-      - name: Install
-        if: steps.node-modules-cache.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+      - if: steps.node-modules-cache.outputs.cache-hit != 'true'
+        run: |
+          yarn install --frozen-lockfile
+          yarn run lint:lockfile
+          yarn deps
 
-      - name: Lint lock file
-        run: yarn run lint:lockfile
-
-  tests:
+  lint-tests:
     needs: [install-dependencies]
-    # Skip this job if the head commit’s message contains the string “[skip ci]”.
-    if: "!contains(github.event.head_commit.message, '[skip ci]')"
-    name: 'Tests'
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout repository
-        # https://github.com/actions/checkout
-        uses: actions/checkout@v3
-
-      - name: Setup Node.js
-        # https://github.com/actions/setup-node
-        uses: actions/setup-node@v3
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
         with:
           node-version-file: '.nvmrc'
-
-      - name: Cache node_modules
-        # https://github.com/actions/cache
-        uses: actions/cache@v3
+      - uses: actions/cache@v3
         id: node-modules-cache
         with:
           path: |
             **/node_modules
+            /home/runner/.cache
           key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
-
-      - name: Lint
-        run: yarn run lint
-        env:
+      - env:
           LINTER_MODE: strict
+        run: |
+          yarn run lint
+          yarn run lint:ts
 
-      - name: Lint TS
-        run: yarn run lint:ts
+  cli-tests:
+    needs: [install-dependencies]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+      - uses: actions/cache@v3
+        id: node-modules-cache
+        with:
+          path: |
+            **/node_modules
+            /home/runner/.cache
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+      - run: |
+          yarn run test
 
-      - name: Test
-        run: yarn run test
+  browser-tests:
+    needs: [install-dependencies]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+      - uses: actions/cache@v3
+        id: node-modules-cache
+        with:
+          path: |
+            **/node_modules
+            /home/runner/.cache
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+      - run: |
+          yarn build:staging
+      - run: |
+          yarn preview:built &
+          yarn run test:browser
+
+  post-checks:
```

Comment:
**Suggestion**: Instead of this job, why don’t we simply add the three jobs to the branch protection rule? Then we could remove this entire job. It only needs those jobs to be named.

Reply:
Because doing that is in a web gui rather than in code.

We can't avoid adding a single branch protection rule in a web gui as far as I know. But if we are changing the number of these from now and in the future, it means its all tracked in git in code with history etc etc.

I'm still fiddling with this unfortunately, and I think its possible its just getting it right. If for whatever reason I find its not possible then we'll have to resort to adding these via a web gui.